h/models/annotation_moderation.py

```diff
@@ -25,7 +25,8 @@ class AnnotationModeration(Base, Timestamps):
                               nullable=False)
 
     #: The annotation which has been flagged.
-    annotation = sa.orm.relationship('Annotation')
+    annotation = sa.orm.relationship('Annotation',
+                                     backref=sa.orm.backref('moderation', uselist=False, cascade='all, delete, delete-orphan'))
```

Comment:
Strictly, the presence of `ondelete='cascade'` in the FK defined above means that the `delete` cascade is redundant here (and potentially less efficient than letting the database do the work). What we probably want to do is:

- leave the database-side cascade as-is
- use `cascade='all, delete-orphan'` in the backref
- add `passive_deletes` to the backref, which will ensure that SQLA won't force the loading of `AnnotationModeration` rows when deleting annotations.

See [the call-out box in the SQLA docs for more information](http://docs.sqlalchemy.org/en/rel_1_1/orm/cascades.html#delete).